lms/djangoapps/shoppingcart/views.py

```diff
@@ -381,7 +381,18 @@ def register_code_redemption(request, registration_code):
                         kwargs['mode'] = course_registration.mode_slug
                     else:
                         raise RedemptionCodeError()
-                redemption.course_enrollment = CourseEnrollment.enroll(request.user, course.id, **kwargs)
+                # If the user already has an enrollment for this course, update the enrollment rather than
+                # creating a new one.
```

Comment:
I'm not sure why this is necessary -- isn't `CourseEnrollment.enroll()` idempotent?